homeassistant/components/sensor/sensehat.py

```diff
@@ -0,0 +1,104 @@
+"""
+A component which show the value of temperature, humidity and pressure
+from Sense HAT board in the form of platform with graphs in history.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/
+"""
+
+import os
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+from sense_hat import SenseHat
+
+
+def get_cpu_temp():
+    # get CPU temperature 
+    res = os.popen("vcgencmd measure_temp").readline()
+    t = float(res.replace("temp=", "").replace("'C\n", ""))
+    return t
+
+
+def get_average(x):
+    # use moving average to get better readings
+    if not hasattr(get_average, "t"):
+        get_average.t = [x, x, x]
+    get_average.t[2] = get_average.t[1]
+    get_average.t[1] = get_average.t[0]
+    get_average.t[0] = x
+    xs = (get_average.t[0]+get_average.t[1]+get_average.t[2])/3
+    return xs
+
+sense = SenseHat()
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    add_devices([SensehatSensor_temperature()])
+    add_devices([SensehatSensor_humidity()])
+    add_devices([SensehatSensor_pressure()])
+
+
+class SensehatSensor_temperature(Entity):
+    #Representation of a  Temperature Sensor.
+
+    @property
+    def name(self):
+    # Return the name of the sensor.
+        return 'Temperature'
+
+    @property
+    def state(self):
+    # Return state of the sensor
+        t1 = sense.get_temperature_from_humidity()
+        t2 = sense.get_temperature_from_pressure()
+        t_cpu = get_cpu_temp()
```

Comment:
You should fetch all the new data inside an `update` method. Within properties you should always only retrieve cached values